doc/website/blog/2024-02-02-atlas-v-0-19.mdx

```diff
@@ -0,0 +1,364 @@
+you can implement complex business logic in your database, without having to write
+any additional code in your application.
+
+Managing triggers as part of the software development lifecycle can be quite a challenge.
+Luckily, Atlas's database schema-as-code approach makes it easy to do!
+
+:::info BETA FEATURE
+Triggers are currently in beta and available to logged-in users only. To use this feature, run:
+
+```
+atlas login
+```
+:::
+
+Let's use Atlas to build a small chunk of a simple e-commerce application:
+
+1. Download the latest version of the Atlas CLI:
+   <InstallationInstructions/>
+2. Make sure you are logged in to Atlas:
+   ```
+   atlas login
+   ```
+3. Let's spin up a new SQL Server database using `docker`:
+
+    ```bash
+    docker run --rm -e 'ACCEPT_EULA=Y' -e 'MSSQL_SA_PASSWORD=P@ssw0rd0995' -p 1433:1433 --name atlas-demo -d mcr.microsoft.com/mssql/server:latest
+    ```
+
+4. Next, let's define and apply the base table for our application:
+
+   ```hcl title=schema.hcl
+   schema "dbo" {
+   }
+    table "grades" {
+      schema = schema.dbo
+      column "student_id" {
+        null = false
+        type = bigint
+      }
+      column "course_id" {
+        null = false
+        type = bigint
+      }
+      column "grade" {
+        null = false
+        type = int
+      }
+      column "grade_status" {
+        null = true
+        type = varchar(10)
+      }
+      primary_key {
+        columns = [column.student_id, column.course_id]
+      }
+    }
+    ```
+    The `grades` table represents a student's grade for a specific course. The column `grade_status` will remain null
+    at first, and we will use a trigger to update whether it the grade is pass or fail.
+
+    Apply this schema on our local SQL Server instance using the Atlas CLI:
+
+    ```bash
+    atlas schema apply \
+      --url "sqlserver://sa:P@ssw0rd0995@localhost:1433?database=master" \
+      --to "file://schema.hcl" \
+      --dev-url "docker://sqlserver/2022-latest/dev?mode=schema" \
+      --auto-approve
+    ```
+
+    This command will apply the schema defined in `schema.hcl` to the local SQL Server instance.
+    Notice the `--auto-approve` flag, which instructs Atlas to automatically apply the schema
+    without prompting for confirmation.
+
+5. Now, let's define the logic to assign a `grade_status` using a `TRIGGER`. Append
+   this definition to `schema.hcl`:
+
+   ```hcl title=schema.hcl (continued)
+    trigger "after_grade_insert" {
+      on = table.grades
+      after {
+        insert = true
+      }
+      as = <<-SQL
+       BEGIN
+         SET NOCOUNT ON;
+
+        UPDATE grades
+        SET grade_status = CASE
+            WHEN inserted.grade >= 70 THEN 'Pass'
+            ELSE 'Fail'
+            END
+            FROM grades
+        INNER JOIN inserted ON grades.student_id = inserted.student_id and grades.course_id = inserted.course_id;
+        END
+       SQL
+      }
+    ```
```

Comment:
Spacing fix
```suggestion
```hcl title=schema.hcl (continued)
trigger "after_grade_insert" {
  on = table.grades
  after {
    insert = true
  }
  as = <<-SQL
    BEGIN
      SET NOCOUNT ON;

    UPDATE grades
    SET grade_status = CASE
        WHEN inserted.grade >= 70 THEN 'Pass'
        ELSE 'Fail'
        END
        FROM grades
    INNER JOIN inserted ON grades.student_id = inserted.student_id and grades.course_id = inserted.course_id;
    END
    SQL
  }
```
```